substrate/client/transaction-pool/tests/integration.rs

```diff
@@ -133,3 +129,61 @@ async fn send_future_and_ready_from_many_accounts_to_relaychain() {
 	assert_eq!(finalized_future, 10_000);
 	assert_eq!(finalized_ready, 10_000);
 }
+
+// Test which sends 5m transactions to parachain. Long execution time expected.
+#[tokio::test(flavor = "multi_thread")]
+#[ignore]
+async fn send_5m_from_many_accounts_to_parachain() {
+	let net = NetworkSpawner::from_toml_with_env_logger(para::HIGH_POOL_LIMIT_FATP)
+		.await
+		.unwrap();
+
+	// Wait for the parachain collator to start block production.
+	net.wait_for_block_production("charlie").await.unwrap();
+
+	// Create future & ready txs executors.
+	let ws = net.node_rpc_uri("charlie").unwrap();
+	let executor = default_zn_scenario_builder(&net)
+		.with_rpc_uri(ws.clone())
+		.with_start_id(0)
+		.with_last_id(999)
+		.with_txs_count(5_000)
+		.with_executor_id("txs-executor".to_string())
+		.build()
+		.await;
+
+	// Execute transactions and fetch the execution logs.
+	let execution_logs = executor.execute().await;
+	let finalized_txs = execution_logs.values().filter_map(|tx_log| tx_log.finalized()).count();
+
+	assert_eq!(finalized_txs, 5_000_000);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[ignore]
+async fn gossiping() {
```

Comment:
nit: not super sure what aspect of the gossiping is the test checking. I saw the network has many validators, but maybe we want to check something specific to gossping, or simply asserting on 1k finalized txs is enough? Maybe a short comment for the purpose of this test can help clarifying its purpose.